packages/mgt-chat/src/utils/onRenderMessage.tsx

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * Props for an adaptive card message.s
+ */
+interface MGTAdaptiveCardProps {
+  attachments: ChatMessageAttachment[];
+  defaultOnRender?: (props: MessageProps) => JSX.Element;
+  messageProps: MessageProps;
+}
+
+/**
+ * Render an adaptive card from the attachments
+ */
+const MGTAdaptiveCard = (msg: MGTAdaptiveCardProps) => {
+  const cardRef = useRef<HTMLDivElement | null>(null);
+  const attachments = msg.attachments;
+  const adaptiveCardAttachment = getAdaptiveCardAttachment(attachments);
+  useEffect(() => {
+    if (adaptiveCardAttachment) {
+      const cardHtmlElement = getHtmlElementFromAttachment(adaptiveCardAttachment);
+      cardRef?.current?.appendChild(cardHtmlElement!);
+    }
+  }, [cardRef, adaptiveCardAttachment]);
+  const defaultOnRender = msg?.defaultOnRender;
+  const messageProps = msg.messageProps;
+  const defaultRender = defaultOnRender ? defaultOnRender(messageProps) : <></>;
+  return adaptiveCardAttachment ? <div ref={cardRef}></div> : defaultRender;
+};
+
+/**
+ * Renders the preferred content depending on whether it is supported.
+ *
+ * @param messageProps final message values from the state.
+ * @param defaultOnRender default component to render content.
+ * @returns
+ */
+const onRenderMessage = (messageProps: MessageProps, defaultOnRender?: MessageRenderer) => {
+  const message = messageProps?.message;
+  if (isGraphChatMessage(message)) {
+    const attachments = message?.attachments ?? [];
+
+    if (message?.hasUnsupportedContent) {
+      const unsupportedContentComponent = <UnsupportedContent targetUrl={message.rawChatUrl} />;
+      messageProps = produce(messageProps, (draft: MessageProps) => {
+        if (isChatMessage(draft.message)) {
+          draft.message.content = renderToString(unsupportedContentComponent);
+        }
+      });
+    } else if (attachments.length) {
+      return (
+        <MGTAdaptiveCard attachments={attachments} defaultOnRender={defaultOnRender} messageProps={messageProps} />
+      );
+    }
+  }
+  return defaultOnRender ? defaultOnRender(messageProps) : <></>;
+};
+
+/**
+ * Filters out the adaptive card attachments.
+ * @param attachments
+ * @returns
+ */
+const getAdaptiveCardAttachment = (attachments: ChatMessageAttachment[]): ChatMessageAttachment | undefined => {
+  for (const att of attachments) {
+    const contentType = att?.contentType ?? '';
+    if (contentType === 'application/vnd.microsoft.card.adaptive') {
+      return att;
+    }
+  }
+  return;
+};
+
+/**
+ * Process the attachment object and return an HTMLElement or nothing.
+ * @param attachment
+ * @returns
+ */
+const getHtmlElementFromAttachment = (attachment: ChatMessageAttachment | undefined): HTMLElement | undefined => {
+  /* eslint-disable
+    @typescript-eslint/no-unsafe-assignment,
+    @typescript-eslint/no-unsafe-member-access,
+    @typescript-eslint/no-unsafe-call, react-hooks/rules-of-hooks */
+  const adaptiveCard = new AdaptiveCards.AdaptiveCard();
```

Comment:
With the selective import we don't need to dereference to the class we need.
```suggestion
  const adaptiveCard = new AdaptiveCard();
```